app/user.go

```diff
@@ -981,6 +981,51 @@ func (a *App) UpdateActive(c request.CTX, user *model.User, active bool) (*model
 		}
 	}
 
+	dmChannels := []*model.Channel{}
+	channelsInTeam, aErr := a.GetChannelsForTeamForUser(c, "", user.Id, &model.ChannelSearchOpts{
+		IncludeDeleted: true,
+		LastDeleteAt:   0,
+	})
+	// soft errors here on since user is updated
+	if aErr != nil {
+		c.Logger().Error("Failed to get channels for user", mlog.String("user_id", user.Id), mlog.Err(err))
+	}
+
+	if channelsInTeam != nil {
+		for _, channel := range channelsInTeam {
+			if channel.Type == model.ChannelTypeDirect {
+				dmChannels = append(dmChannels, channel)
```

Comment:
Since our below operations are operating on atomic channels, we could skip the extra step of creating the `dmChannels` slice and simply `continue` if it's a condition we don't care about:

```go
if channel.Type != model.ChannelTypeDirect {
    continue
}
```